concordium-contracts-common/src/schema.rs

```diff
@@ -780,6 +874,76 @@ impl Deserial for FunctionV2 {
     }
 }
 
+impl Serial for FunctionV3 {
```

Comment:
If we want to just use FunctionV2 for both V2 and V3 contract schemas, and we want the new and improved 3 `Option`s version, then we need to make it compatible with the existing v2 serialization. I.e.:

0 => param
1 => rv
 2 => param + rv
3 => error
4 => param + error
5 => rv + error
6 => param + rv + error